pkg/db/db.go

```diff
@@ -0,0 +1,413 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkgdb
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+)
+
+const ImplicitColName = "_tidb_rowid"
+const ImplicitColID = -1
+
+// CloseDB close the mysql fd
+func CloseDB(db *sql.DB) error {
+	return errors.Trace(db.Close())
+}
+
+// GetCreateTable gets the create table sql.
+func GetCreateTable(db *sql.DB, schemaName string, tableName string) (string, error) {
+	query := fmt.Sprintf("SHOW CREATE TABLE `%s`.`%s`", schemaName, tableName)
+	row := db.QueryRow(query)
+
+	var tbl, createTable sql.NullString
+	err := row.Scan(&tbl, &createTable)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if !tbl.Valid || !createTable.Valid {
```

Comment:
need to check `tbl ==tableName`?

Reply:
What are the benefits of verify `tbl.Valid`？

Reply:
for example, db1 has table test, and db2 don't have, the result of show create table for db2 is nil, if don't judge valid, may cause panic